# Benchmarking gene embeddings from sequence, expression, network, and text models for functional prediction tasks
This repository contains code and datasets for benchmarking 38 gene embedding methods across functional prediction tasks, including gene attribute, interaction, and set relationship predictions, using data from amino acid sequences, gene expression, PPI networks, and biomedical literature.

## Organization
This repo is organized into several sections, part of which is stored on zenodo.
- `data`: contains ... on zenodo
- `src`: contains the code used for related tasks to preprocess the embeddings and benchmakr them across our three main functional prediction tasks

## Installation
To install the required dependencies for the project, use `pip` to install the dependencies listed in the `requirements.txt` file. Ensure that your Python version is 3.8 or higher.
```bash
pip install -r requirements.txt
```
The requirements.txt file includes all necessary dependencies, which will be installed automatically when you run the above command.
